chainer/gradient_check.py

```diff
@@ -622,30 +631,47 @@ def perturb(data, direction):
             # Input arrays
             g_x_vars = []
             j = 0
-            for i in range(len(x_vars)):
-                if no_grads[i]:
-                    g_x_vars.append(x_vars[i])
+            for x_var, skip in six.moves.zip(x_vars, no_grads):
+                if skip:
+                    g_x_vars.append(x_var)
                 else:
                     data = perturb(casted_data[j], directions[j])
                     g_x_vars.append(variable.Variable(data))
                     j += 1
+
             # Parameters
-            for i in range(len(params)):
-                params[i].data = perturb(
-                    casted_data[j + i], directions[j + i])
+            for i, param in enumerate(params):
```

Comment:
I think `for i in range(len(params)):` is better.
`param` is not always used.

Reply:
I agree, thanks. Fixed.